File: evals/tasks/observe_simple_google_search.ts
import { EvalFunction } from "@/lib/v3/types/public/evals";

export const observe_simple_google_search: EvalFunction = async ({
  debugUrl,
  sessionUrl,
  v3,
  logger,
}) => {
  try {
    const page = v3.context.pages()[0];
    await page.goto(
      "https://browserbase.github.io/stagehand-eval-sites/sites/google/",
    );
    const observation1 = await v3.observe(
      "Find the search bar and type 'OpenAI'",
    );

    if (observation1.length > 0) {
      const action1 = observation1[0];
      await v3.act(action1);
    }
    const observation2 = await v3.observe(
      "Click the search button in the suggestions dropdown",
    );

    if (observation2.length > 0) {
      const action2 = observation2[0];
      await v3.act(action2);
    }

    const expectedUrl =
      "https://browserbase.github.io/stagehand-eval-sites/sites/google/openai.html";
    const currentUrl = await page.url();

    return {
      _success: currentUrl.startsWith(expectedUrl),
      currentUrl,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } catch (error) {
    return {
      _success: false,
      error: error,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } finally {
    await v3.close();
  }
};

File: packages/sdk-ruby/evals/tasks/observe/observe_simple_google_search.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/observe/observe_simple_google_search.ts

Evals.define_task("observe_simple_google_search") do |t|
  t.page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/google/")

  observation1 = t.stagehand.observe("Find the search bar and type 'OpenAI'").data
  t.stagehand.act(observation1[0]) unless observation1.empty?

  observation2 = t.stagehand.observe("Press enter").data
  t.stagehand.act(observation2[0]) unless observation2.empty?

  t.page.wait_for_timeout(3000)

  expected_url = "https://browserbase.github.io/stagehand-eval-sites/sites/google/openai.html"
  current_url = t.page.url

  { _success: current_url.start_with?(expected_url), currentUrl: current_url }
end
